python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -135,13 +139,37 @@ public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage us
         issue.addQuickFix(quickFix);
       }
       return issue;
+    } else if (usage.kind().equals(Usage.Kind.ASSIGNMENT_LHS)) {
+      PreciseIssue issue = ctx.addIssue(usage.tree(), String.format(MESSAGE, symbol.name()));
+
+      Statement assignmentStatement = ((Statement) TreeUtils.firstAncestorOfKind(usage.tree(), Kind.ASSIGNMENT_STMT, Kind.ANNOTATED_ASSIGNMENT));
+
+      // This can only trigger if there is 1 assigned variable in the same statement, the other cases are handled by isSequenceUnpacking
+
+      Optional.ofNullable(assignmentStatement).ifPresent(stmt -> createAssignmentQuickFix(usage, stmt, issue));
+      return issue;
     } else {
       var issue = ctx.addIssue(usage.tree(), String.format(MESSAGE, symbol.name()));
       createExceptClauseQuickFix(usage, issue);
       return issue;
     }
   }
 
+  private static void createAssignmentQuickFix(Usage usage, Statement assignmentStatement, PreciseIssue issue) {
+    Token lastQuickFixToken;
+    if (assignmentStatement.is(Kind.ASSIGNMENT_STMT)) {
+      lastQuickFixToken = ((AssignmentStatement) assignmentStatement).assignedValue().firstToken();
+    } else {
+      var assignedValue = ((AnnotatedAssignment) assignmentStatement).assignedValue();
+      // This is an assignment, so it can't be null
```

Comment:
Actually, the following:
```
x: str
```
is considered a valid annotated assignment with a null assigned value.

The only reason this is never null in the current test suite is because `hasOnlyBindingUsages` explicitly filters out symbols which are only used as type annotations (through `isOnlyTypeAnnotation`).

In fact modifying one of the existing tests such as:
```
def no_fp_type_annotation():
    value: str  # assigned value is null here
    value = "hello"
```
will make the assert statement fail. 

As a sidenote, I think we should never use `assert` statements to test our assumptions. As these statements may not be executed in production, it is unsafe to rely on them (cf. my comment on the Jira ticket).

Besides, this case is a good example why it is better to let code go uncovered through defensive programming than relying on assumptions. With an uncovered `null` check, we'd miss providing a quick fix in some cases. With an `assert` statement or an unsafe dereference, we'd have a crash in production because the assumption was ultimately incorrect.

(Note: I think we probably need an extra test case that reflects the example here)